include/numerics/tensor_tools.h

```diff
@@ -23,6 +23,10 @@
 // Local includes
 #include "libmesh/libmesh_common.h"
 #include "libmesh/compare_types.h"
+#ifdef LIBMESH_HAVE_METAPHYSICL
+#include "metaphysicl/dualnumber.h"
+#include "metaphysicl/dynamicsparsenumberbase.h"
+#endif
```

Comment:
Is this the right way to do things?  tensor_tools.h is included by both type_vector.h and dense_vector.h so we'd basically be adding a metaphysicl dependency to nearly every file in libMesh and MOOSE!

I know it's a pain to have to include multiple sets of headers (and in the right order, even if we need go so far as to use _decl headers to handle nesting) to do generic programming with their combination, but it's better than having everything depend on everything else - I'd even go so far as to make that norm_sq case take ```const T & a``` so you don't need reference to DualNumber at all there.